survey/src/main/java/com/quickbirdstudios/surveykit/backend/address/AddressSuggestionProvider.kt

```diff
@@ -0,0 +1,34 @@
+package com.quickbirdstudios.surveykit.backend.address
+
+import com.quickbirdstudios.surveykit.AnswerFormat
+
+abstract class AddressSuggestionProvider(
```

Comment:
This does not need to be an abstract class. You can simply make it an interface
```kotlin
interface AddressSuggestionProvider{
    val onSuggestionListReady: (suggestions: List<AddressSuggestion>) -> Unit?

    fun input(query: String)

    data class AddressSuggestion(
```

and then 
```kotlin
class YandexAddressSuggestionProvider(
    private val apiKey: String,
    private val resultsCount: Int = 5,
    private val lang: String = "tr_TR",
    override val onSuggestionListReady: (suggestions: List<AddressSuggestion>) -> Unit? = {}
) : AddressSuggestionProvider {
```